### PR TITLE
enh(phpstan) : fix phpstan errors in centreon/src (part 4)

### PR DESCRIPTION
**Pull request by AssanDiomande**
## Description

**Fixes** # MON-11920

updated files in :

* src/CentreonModule

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

### CODE REVIEW

File: src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php [hunk 1]
```diff
@@ -148,101 +177,107 @@ public function tearDown(): void
...
     }
 
-    public function testCreateEntityFromConfig()
+    /**
+    * @return void
+    */
+    public function testCreateEntityFromConfig(): void
     {
-        $configFile = static::getConfFilePath();
+        $configFile = self::getConfFilePath();
```
Comment (callapa) [inline src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php:242]: Why are you changing the static keyword to self?

Comment (AssanDiomande) [reply src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php:242]: It's a mistake I will fix in the next push cause I'm fixing some errors noted on jenkins

**Review by kduret — DISMISSED** [review]

File: src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php (hunk 1 above)
Comment (AssanDiomande) [reply src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php]: reverted

File: src/CentreonModule/Infrastructure/Entity/Module.php [hunk 2]
```diff
@@ -97,7 +97,7 @@ class Module implements SourceDataInterface
     private $keywords;
 
     /**
-     * @var string
+     * @var string|array<string,string|bool>
```
Comment (a-launois) [inline src/CentreonModule/Infrastructure/Entity/Module.php:100]:
```suggestion
     * @var array<string,string|bool>
```

File: src/CentreonModule/Application/DataRepresenter/ModuleDetailEntity.php [hunk 3]
```diff
@@ -80,7 +80,7 @@ public function __construct(Module $entity)
      *
      * JSON serialization of entity
      *
-     * @return array
+     * @return array<string,mixed| array<string,string|bool|null>>
```
Comment (callapa) [inline src/CentreonModule/Application/DataRepresenter/ModuleDetailEntity.php:83]:
```suggestion
     * @return array<string, mixed>
```

File: src/CentreonModule/Application/DataRepresenter/ModuleEntity.php [hunk 4]
```diff
@@ -75,7 +75,7 @@ public function __construct(Module $entity)
      *
      * JSON serialization of entity
      *
-     * @return array
+     * @return array<string,mixed| array<string,string|bool|null>>
```
Comment (callapa) [inline src/CentreonModule/Application/DataRepresenter/ModuleEntity.php:78]:
```suggestion
     * @return array<string, mixed>
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php [hunk 5]
```diff
@@ -133,11 +162,11 @@ protected function setUp(): void
         $this->source
             ->method('getModuleConf')
             ->will($this->returnCallback(function () {
-                    $result = [
+                $result = [
                         ModuleSourceTest::$moduleName => ModuleSourceTest::$moduleInfo,
                     ];
```
Comment (a-launois) [inline src/CentreonModule/Tests/Infrastructure/Source/ModuleSourceTest.php:167]:
```suggestion
                $result = [
                    ModuleSourceTest::$moduleName => ModuleSourceTest::$moduleInfo,
                ];
```

**Review by a-launois — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**